packages/flutter_test/lib/src/binding.dart

```diff
@@ -646,12 +675,20 @@ class AutomatedTestWidgetsFlutterBinding extends TestWidgetsFlutterBinding {
       testBodyResult = _runTest(testBody, invariantTester, description);
       assert(inTest);
     });
-    // testBodyResult is a Future that was created in the Zone of the fakeAsync.
-    // This means that if we call .then() on it (as the test framework is about to),
-    // it will register a microtask to handle the future _in the fake async zone_.
-    // To avoid this, we wrap it in a Future that we've created _outside_ the fake
-    // async zone.
-    return new Future<Null>.value(testBodyResult);
+
+    return new Future<Null>.microtask(() async {
+      while (_pendingAsyncTasks != null) {
```

Comment:
you need to flush microtasks at least once even when it's null

Reply:
oh i see, you do that in the async barrier.

Yeah, that makes sense.

Reply:
> oh i see, you do that in the async barrier.

I thought so too, which is why I didn't put it in here.  But your first instinct was right - we do need it, because the work done in `runAsync()` may have just completed, meaning `_pendingAsyncTasks` will be null, yet the microtask queue is still backed up.